### PR TITLE
Fix issue of missing components in output

### PR DESCRIPTION
**Pull request by kavitharaju**
This PR includes
- More tests and fixes in the context of #192 
   All markers in the input USFM are found using regex pattern and the outputs 
       * SyntaxTree, 
       * dict and 
       * USX 
    are searched for these markers to ensure all usfm components are represented in parsed output formats
- A grammar change to combine adjacent poetry elements within one "poetry" element in syntax tree and there by in JSON
- Changes in to_list() 
   - To accommodate for the change in dict for using `v` and `c` as keys instead of chapterNumber and verseNumber
   - Tests to make sure conversions happen without failure
   - Minor fixes to handle some failures
   - This methods needs to be re-written to work with the flat JSON schema. So, no further changes or tests done for now.
- A small change in Exception handling that was part of previous PR
- Four tests that use \k-s as milestone have been excludeda and noted, as I have doubt about the correctness of this usage.

### CODE REVIEW

File: python-usfm-parser/src/usfm_grammar/usfm_parser.py [hunk 1]
```diff
@@ -57,7 +57,8 @@ class Format(str, Enum):
 TABLE_CELL_MARKERS = ["tc", "th", "tcr", "thr"]
 
 ANY_VALID_MARKER = PARA_STYLE_MARKERS+NOTE_MARKERS+CHAR_STYLE_MARKERS+\
-                    NESTED_CHAR_STYLE_MARKERS+TABLE_CELL_MARKERS
+                    NESTED_CHAR_STYLE_MARKERS+TABLE_CELL_MARKERS+\
+                    ["fig", "cat", "esb", "b", "ph", "pi"]
```
Comment (joelthe1) [inline python-usfm-parser/src/usfm_grammar/usfm_parser.py:61]: Any reason why these markers are not categorized like the rest?

Comment (kavitharaju) [reply python-usfm-parser/src/usfm_grammar/usfm_parser.py:61]: First 4 could be called special cases maybe. Pi and pH are para markers but with intendation. So in short "special" would be the only term to call them collectively. Nothing much in common.

Comment (joelthe1) [reply python-usfm-parser/src/usfm_grammar/usfm_parser.py:61]: Okay, so maybe calling them `MISC_MARKERS` (for miscellaneous) or something such?

Comment (kavitharaju) [reply python-usfm-parser/src/usfm_grammar/usfm_parser.py]: Changed :+1:

**Pull request merged**